### PR TITLE
ci: add nightly fuzzing job

### LINKED ISSUE #169: test: run a nightly CI with increased fuzzing
## Why

More and more tests purely rely on fuzzing and diff testing against the reference EELS implementation. While we don't want to slow down the dev cycle, we want to make sure that we don't let cracks in the code.

## What

Add a nightly job like similar to https://github.com/kkrt-labs/kakarot/blob/main/.github/workflows/cairo-zero-nightly-fuzzing.yml

### PR DESCRIPTION
**Pull request by enitrat**
Closes #169

### CODE REVIEW

File: .github/workflows/nighly_fuzzing.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+          version: nightly
+      - name: Run tests
+        id: tests
+        continue-on-error: true # To send a notification if the tests fail
+        run: |
+          echo "HYPOTHESIS_PROFILE=nightly" >> .env
+          cd cairo
+          uv run compile
+          forge build
+          uv run pytest "tests" -m "not NoCI" -n logical --seed 42 --junitxml=junit.xml -o junit_family=legacy
```
Comment (obatirou) [inline .github/workflows/nighly_fuzzing.yml:37]: I don't know if we want/need to have a seed 
Does it influence hypothesis input generation ?

File: .github/workflows/nighly_fuzzing.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - uses: actions/setup-python@v5
+        with:
+          python-version-file: ./cairo/.python-version
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+      - name: Run tests
+        id: tests
+        continue-on-error: true # To send a notification if the tests fail
```
Comment (obatirou) [inline .github/workflows/nighly_fuzzing.yml:31]: Seems to make the step green even if failed

Comment (enitrat) [reply .github/workflows/nighly_fuzzing.yml:31]: yes, hence the

```
      - name: Fail workflow if tests failed
        if: steps.tests.outcome == 'failure'
        run: exit 1
```

at the end

File: .github/workflows/nighly_fuzzing.yml (hunk 1 above)
Comment (enitrat) [reply .github/workflows/nighly_fuzzing.yml:37]: I think it only determines the seed for the random number generators, not the test themselves. you're right that it's probably best without.

**Review by ClementWalter — APPROVED** [review]

**Pull request merged**